test/jest/persistence.spec.ts

```diff
@@ -0,0 +1,145 @@
+            },
+        } as unknown as IMParticleWebSDKInstance;
+
+        store = {} as IStore;
+        Store.call(store, {} as SDKInitConfig, mockMPInstance, 'apikey');
+        window.localStorage.removeItem('mprtcl-v4');
+    });
+
+    describe('#noFunctional privacy flag', () => {
+        describe('true', () => {
+            beforeEach(() => {
+                store.setNoFunctional(true);
+                store.webviewBridgeEnabled = false;
+            });
+
+            it('should NOT write to cookie and localStorage when useCookieStorage is true', () => {
+                store.SDKConfig.useCookieStorage = true;
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).not.toHaveBeenCalled();
+                expect(window.localStorage.getItem('mprtcl-v4')).toBeNull();
+            });
+
+            it('should NOT write to localStorage when useCookieStorage is false', () => {
+                store.SDKConfig.useCookieStorage = false;
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).not.toHaveBeenCalled();
+                expect(window.localStorage.getItem('mprtcl-v4')).toBeNull();
+            });
+        });
+
+        describe('false', () => {
+            beforeEach(() => {
+                store.setNoFunctional(false);
+                store.webviewBridgeEnabled = false;
+            });
+
+            it('should write to cookie and localStorage when useCookieStorage is true', () => {
+                store.SDKConfig.useCookieStorage = true;
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).toHaveBeenCalled();
+                expect(window.localStorage.getItem('mprtcl-v4')).not.toBeNull();
+            });
+
+            it('should write to localStorage when useCookieStorage is false', () => {
+                store.SDKConfig.useCookieStorage = false;
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).toHaveBeenCalled();
+                expect(window.localStorage.getItem('mprtcl-v4')).not.toBeNull();
+            });
+        });
+
+        describe('is false by default', () => {
+            beforeEach(() => {
+                // default is false
+                store.webviewBridgeEnabled = false;
+            });
+
+            it('should write to cookie and localStorage by default when useCookieStorage is true', () => {
```

Comment:
Are you sure it writes to both localStorage AND cookie storage? It should only be one or the other?